parsl/executors/taskvine/executor.py

```diff
@@ -114,80 +98,90 @@ class TaskVineExecutor(BlockProviderExecutor, putils.RepresentationMixin):
     @typeguard.typechecked
     def __init__(self,
                  label: str = "TaskVineExecutor",
-                 use_factory: bool = False,
+                 worker_launch_method: str = 'factory',
                  manager_config: TaskVineManagerConfig = TaskVineManagerConfig(),
                  factory_config: TaskVineFactoryConfig = TaskVineFactoryConfig(),
                  provider: Optional[ExecutionProvider] = LocalProvider(init_blocks=1),
                  storage_access: Optional[List[Staging]] = None):
 
-        # If TaskVine factory is used, disable the Parsl provider
-        if use_factory:
+        # Set worker launch option for this executor
+        if worker_launch_method == 'factory' or worker_launch_method == 'manual':
             provider = None
+        elif worker_launch_method != 'provider':
+            raise ExecutorError(self, "Worker launch option '{worker_launch_method}' \
+                                       is not supported.")
```

Comment:
i find this style of `if` statement a bit awkward to reason about in my head, about are all the cases of worker_launch_method covered here, and prefer the style that is:

if a = X or a = Y:
  some thing
elif a = Z:
  some other thing which might be `pass`
else:
  raise failure

that's something of a personal preference though